File: bookexcel/template/PHP/libs/ConfigItemBase.php
<?php
/**
 * Copyright (c) 2016, bookrpg, All rights reserved.
 * @author llj <wwwllj1985@163.com>
 * @license The MIT License
 */

include_once  __DIR__ . '/IConfigParser.php';

class ConfigItemBase
{
    protected $_key1;
    protected $_key2;

    public function parseFrom(IConfigParser $parser)
    {
        throw new Exception('not implements', 1);
    }

    public function getKey()
    {
        return $this->_key1;
    }

    public function getSecondKey()
    {
        return $this->_key2;
    }
}

File: bookexcel/template/C#/libs/ConfigItemBase.cs
﻿///
/// Copyright (c) 2016, bookrpg, All rights reserved.
/// @author llj <wwwllj1985@163.com>
/// @license The MIT License
///

using System;
using System.Collections;

namespace bookrpg.config
{
    public abstract class ConfigItemBase
    {
        protected object _key1;
        protected object _key2;

        public ConfigItemBase()
        {
        }

        public virtual bool ParseFrom(IConfigParser parser)
        {
            throw new NotImplementedException();
        }

        public virtual object GetKey()
        {
            return _key1;
        }

        public virtual object GetSecondKey()
        {
            return _key2;
        }
    }
}